woocommerce/woocommerce-ios
# Add CustomerStore and support for loadCustomer action

Part of https://github.com/woocommerce/woocommerce-ios/issues/7741

### Description

This PR adds a `CustomerStore` and `CustomerAction` to Yosemite, as well as a temporary implementation of a `loadCustomer` action, which returns a `Result<Customer, Error>` type to a completion handler.

It also adds a "Magnifying Glass" button to the "Add Customer" screen within Order Creation, hidden behind a feature flag.

### Changes: 

- Adds `CustomerStore` to Yosemite Stores.
- Adds `CustomerAction` to Yosemite Actions.
- Adds `Networking.Customer` typealias to Yosemite
- Adds a "magnifying glass" button to "Orders" > "+" > "Create new Order" > "+ Add Customer" that calls the Customer action.

### Testing instructions:

The current implementation assumes (temporarily, the customerID logic will be part of a different PR) that you have at least one customer in your store, and that customerID is `1`, if that's not the case, you can change this line to the customerID you prefer:

https://github.com/woocommerce/woocommerce-ios/blob/a8736d4c7c30a8c4d41d27854a16e908d553f082/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/CustomerSection/CreateOrderAddressFormViewModel.swift#L108

How can you check your customers and customerID's? The quick way is going to WooCommerce > Customers > Hover over any of the customers and check the bottom of the browser:

<img width="700" alt="_get customer ID" src="https://user-images.githubusercontent.com/3812076/192416322-3b14698c-5389-4ae4-9561-eaf25b80dc93.png">

- Go to "Orders" > "+" > "Create new Order" > "+ "Add Customer" and tap in the "Magnifying glass" icon at the top of the screen
<img width="439" alt="Screenshot 2022-09-26 at 13 11 23" src="https://user-images.githubusercontent.com/3812076/192192292-26a940ff-353f-4c1b-99ba-54d443c0a91d.png">

- Confirm in Proxyman or any other tool that we send a GET HTTP request to `/wc/v3/customers/{your-customerID}` and that returns a 200 response, example:

```
https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/121710041/rest-api/?json=true&path=/wc/v3/customers/1%26_method%3Dget
```
```
{
  "data": {
    "id": 1,
    "date_created": "2016-12-25T16:05:38",
    "date_created_gmt": "2016-12-25T16:05:38",
    "date_modified": "2022-09-13T08:30:03",
    "date_modified_gmt": "2022-09-13T08:30:03",
    "email": "gabriel.maldonado@automattic.com",
    "first_name": "GAB",
    "last_name": "MAL",
    "role": "administrator",
... 
}
```
